src/Sarif.Converters.UnitTests/ToolFormatConverterTests.cs

```diff
@@ -109,5 +121,245 @@ public void ToolFormatConverter_TruncatesOutputFileInOverwriteMode()
+            using (var tempDir = new TempDirectory())
+            {
+                const string ToolName = "NonPublicTool";
+                string pluginAssemblyPath = GetCurrentAssemblyPath();
+
+                string inputFilePath = tempDir.Write("input.txt", string.Empty);
+                string outputFilePath = tempDir.Combine("output.txt");
+
+                Action action = () => _converter.ConvertToStandardFormat(
+                    ToolName,
+                    inputFilePath,
+                    outputFilePath,
+                    ToolFormatConversionOptions.None,
+                    pluginAssemblyPath);
+
+                action.ShouldThrow<ArgumentException>()
+                    .Where(ex =>
+                        ex.Message.Contains(pluginAssemblyPath)
+                        && ex.Message.Contains(ToolName));
+            }
+        }
+
+        [TestMethod]
+        public void ToolFileConverter_FailsIfConverterTypeDoesNotHaveCorrectBaseClass()
+        {
+            using (var tempDir = new TempDirectory())
+            {
+                const string ToolName = "IncorrectlyDerivedTool";
+                string pluginAssemblyPath = GetCurrentAssemblyPath();
+
+                string inputFilePath = tempDir.Write("input.txt", string.Empty);
+                string outputFilePath = tempDir.Combine("output.txt");
+
+                Action action = () => _converter.ConvertToStandardFormat(
+                    ToolName,
+                    inputFilePath,
+                    outputFilePath,
+                    ToolFormatConversionOptions.None,
+                    pluginAssemblyPath);
+
+                action.ShouldThrow<ArgumentException>()
+                    .Where(ex =>
+                        ex.Message.Contains(pluginAssemblyPath)
+                        && ex.Message.Contains(ToolName));
+            }
+        }
+
+        [TestMethod]
+        public void ToolFileConverter_FailsIfConverterTypeDoesNotHaveDefaultConstructor()
+        {
+            using (var tempDir = new TempDirectory())
+            {
+                const string ToolName = "NoDefaultConstructorTool";
+                string pluginAssemblyPath = GetCurrentAssemblyPath();
+
+                string inputFilePath = tempDir.Write("input.txt", string.Empty);
+                string outputFilePath = tempDir.Combine("output.txt");
+
+                Action action = () => _converter.ConvertToStandardFormat(
+                    ToolName,
+                    inputFilePath,
+                    outputFilePath,
+                    ToolFormatConversionOptions.None,
+                    pluginAssemblyPath);
+
+                action.ShouldThrow<ArgumentException>()
+                    .Where(ex =>
+                        ex.Message.Contains(pluginAssemblyPath)
+                        && ex.Message.Contains(ToolName));
+            }
+        }
+
+        [TestMethod]
+        public void ToolFormatConverter_FindsConverterInPluginAssembly()
```

Comment:
>ToolFormatConverter_FindsConverterInPluginAssembly [](start = 20, length = 50)

File access tests generally require additional test cases to be complete. What about UnauthorizedAccessException, for example, if someone provides a path to a dll where access isn't granted? Also BadImageFormatException is another weird one that doesn't extend IOException. Speaking of which, do you have a test with illegal characters in path to produce an IOException?? Looking now. :)